### ISSUE TITLE
Upgrade from 1.12.9 is broken

### ISSUE THREAD

**Issue by CecileRobertMichon** [Azure/aks-engine#1421]
Repro:

{   "apiVersion": "vlabs",   "properties": {     "orchestratorProfile": {       "orchestratorType": "Kubernetes"     },     "masterProfile": {       "count": 1,       "dnsPrefix": "",       "vmSize": "Standard_D2_v2"     },     "agentPoolProfiles": [       {         "name": "agentpool1",         "count": 3,         "vmSize": "Standard_D2_v2",         "osType": "Linux"       }     ],     "linuxProfile": {       "adminUsername": "azureuser",       "ssh": {         "publicKeys": [           {             "keyData": ""           }         ]       }     },     "servicePrincipalProfile": {       "clientId": "",       "secret": ""     }   } }

from commit aacc7cfea5b37ff4c0e516d69f269b7d03a5ff5e consistently results in: 

```
Error: upgrading cluster: Code="NetworkingInternalOperationError" Message="Unknown network allocation error."
```

**Comment by mo-saeed** [comment]
@CecileRobertMichon Just FYI i got this error sometimes when i was trying to create cluster with version 1.13.6, it happened one or two time then it worked ! I thought it's sth related network maintenance in azure but i am not sure as i didn't investigate.

**Comment by CecileRobertMichon** [comment]
@mo-saeed that was my first intuition too but it's reproing consistently in our tests and doesn't repro with 1.12.8

**Comment by CecileRobertMichon** [comment]
@andyzhangx any ideas?

**Comment by CecileRobertMichon** [comment]
This is reproducible only if a scale happened prior to the upgrade.

**Comment by andyzhangx** [comment]
what's the detailed error of this issue? e.g. node not ready or sth else?

**Comment by CecileRobertMichon** [comment]
it's the error I pasted in the description, ie. `Error: upgrading cluster: Code="NetworkingInternalOperationError" Message="Unknown network allocation error."`

**Comment by andyzhangx** [comment]
I could not repro by following these steps:
1. set up a 1.12.9 cluster with 3 vmss instances
2. scale down to 1 vmss instance
3. upgrade to 1.13.5

I used aks-engine version `678e02a52178d2b02702bf4372e7e66761572dca` on master branch, the upgrade operation completed successfully:
```
 # ./aks-engine upgrade --upgrade-version 1.13.5 -g andy-vmss1129 -s xxx -l eastus2 --client-id xxx --client-secret xxx --api-model _output/andy-vmss1129/apimodel.json --auth-method client_secret
INFO[0002] Upgrading cluster with name suffix: 36736423
INFO[0002] VM k8s-agentpool-36736423-vmss_0 in VMSS k8s-agentpool-36736423-vmss has a current version of 1.12.9 and a desired version of 1.13.5. Upgrading this node.
INFO[0002] Master VM name: k8s-master-36736423-0, orchestrator: 1.12.9 (MasterVMs)
INFO[0002] Upgrading to Kubernetes version 1.13.5
INFO[0002] Master nodes StorageProfile: ManagedDisks
INFO[0002] Prepping master nodes for upgrade...
INFO[0002] Resource count before running NormalizeResourcesForK8sMasterUpgrade: 10
INFO[0002] Evaluating if agent pool: master, resource: [concat(variables('masterVMNamePrefix'), copyIndex(variables('masterOffset')))] needs to be removed
INFO[0002] Evaluating if extension: [concat(variables('masterVMNamePrefix'), copyIndex(variables('masterOffset')),'/cse', '-master-', copyIndex(variables('masterOffset')))] needs to be removed
INFO[0002] Evaluating if extension: [concat(variables('masterVMNamePrefix'), copyIndex(variables('masterOffset')), '/computeAksLinuxBilling')] needs to be removed
INFO[0002] Resource count after running NormalizeResourcesForK8sMasterUpgrade: 10
INFO[0002] Total expected master count: 1
INFO[0002] Master nodes that need to be upgraded: 1
INFO[0002] Master nodes that have been upgraded: 0
INFO[0002] Starting upgrade of master nodes...
INFO[0002] masterNodesInCluster: 1
INFO[0002] Upgrading Master VM: k8s-master-36736423-0
INFO[0002] fetching VM: andy-vmss1129/k8s-master-36736423-0
INFO[0003] found nic name for VM (andy-vmss1129/k8s-master-36736423-0): k8s-master-36736423-nic-0
INFO[0003] deleting VM: andy-vmss1129/k8s-master-36736423-0
INFO[0003] waiting for vm deletion: andy-vmss1129/k8s-master-36736423-0
INFO[0213] deleting nic: andy-vmss1129/k8s-master-36736423-nic-0
INFO[0213] waiting for nic deletion: andy-vmss1129/k8s-master-36736423-nic-0
INFO[0223] deleting managed disk: andy-vmss1129/k8s-master-36736423-0_OsDisk_1_dc13080f17784868b8a0e314d69f8f7b
INFO[0283] Master offset: 0
INFO[0283] Master pool set count to: 1 temporarily during upgrade...
INFO[0283] Starting ARM Deployment (master-19-06-11T05.46.18-351491238). This will take some time...
INFO[0475] Finished ARM Deployment (master-19-06-11T05.46.18-351491238). Succeeded
INFO[0475] Master node: k8s-master-36736423-0 is ready
INFO[0475] Expected master count: 1, Creating 0 more master VMs
INFO[0475] Master node: k8s-master-36736423-0 is ready
INFO[0475] Master node: k8s-master-36736423-0 is ready
INFO[0475] Deploying the agent scale sets ARM template...
INFO[0475] Starting ARM Deployment (agentscaleset-19-06-11T05.49.30-1612528641). This will take some time...
INFO[0486] Finished ARM Deployment (agentscaleset-19-06-11T05.49.30-1612528641). Succeeded
INFO[0486] Upgrading VMSS k8s-agentpool-36736423-vmss
INFO[0486] VMSS k8s-agentpool-36736423-vmss current capacity is 1 and new capacity will be 2 while each node is swapped
INFO[0583] Successfully set capacity for VMSS k8s-agentpool-36736423-vmss
INFO[0583] Draining node k8s-agentpool-36736423-vmss000000
INFO[0583] Node k8s-agentpool-36736423-vmss000000 has been marked unschedulable.
INFO[0583] 4 pods need to be removed/deleted
INFO[0583] tiller-deploy-7bfcdc49d6-n5cxc pod successfully evicted
INFO[0583] heapster-6f6cbcfcf6-qvfwb pod successfully evicted
INFO[0583] metrics-server-67b4964794-klc85 pod successfully evicted
INFO[0583] kubernetes-dashboard-9bf969764-bgc6n pod successfully evicted
INFO[0583] Deleting VM k8s-agentpool-36736423-vmss000000 in VMSS k8s-agentpool-36736423-vmss
INFO[0583] Copying custom annotations, labels, taints from old node k8s-agentpool-36736423-vmss000000 to new node k8s-agentpool-36736423-vmss000003...
INFO[0583] Node k8s-agentpool-36736423-vmss000003 has been marked unschedulable.
INFO[0583] 5 pods need to be removed/deleted
INFO[0583] coredns-59b998c9dd-ktmjj pod successfully evicted
INFO[0583] heapster-6f6cbcfcf6-222nf pod successfully evicted
INFO[0584] metrics-server-67b4964794-v4kt6 pod successfully evicted
INFO[0584] tiller-deploy-7bfcdc49d6-t6l8v pod successfully evicted
INFO[0584] kubernetes-dashboard-9bf969764-9xldb pod successfully evicted
INFO[0585] Successfully copied custom annotations, labels, taints from old node k8s-agentpool-36736423-vmss000000 to new node k8s-agentpool-36736423-vmss000003.
INFO[0705] Successfully deleted VM k8s-agentpool-36736423-vmss000000 in VMSS k8s-agentpool-36736423-vmss
INFO[0705] Completed upgrading VMSS k8s-agentpool-36736423-vmss
INFO[0705] Completed upgrading all VMSS
INFO[0705] Cluster upgraded successfully to Kubernetes version 1.13.5

azureuser@k8s-master-36736423-0:~$ kubectl get no
NAME                                STATUS   ROLES    AGE    VERSION
k8s-agentpool-36736423-vmss000003   Ready    agent    6m8s   v1.13.5
k8s-master-36736423-0               Ready    master   60m    v1.13.5
azureuser@k8s-master-36736423-0:~$ kubectl top nodes
NAME                                CPU(cores)   CPU%   MEMORY(bytes)   MEMORY%
k8s-agentpool-36736423-vmss000003   70m          3%     774Mi           12%
k8s-master-36736423-0               206m         10%    1345Mi          21%
```

@CecileRobertMichon is the error message from aks-engine? you may check the detailed error info from azure activity log

**Comment by CecileRobertMichon** [comment]
@andyzhangx here is the repro (it fails 100% of the time for me, region doesn't matter):

With apimodel:
```
{   "apiVersion": "vlabs",   "properties": {     "orchestratorProfile": {       "orchestratorType": "Kubernetes"     },     "masterProfile": {       "count": 1,       "dnsPrefix": "",       "vmSize": "Standard_D2_v3"     },     "agentPoolProfiles": [       {         "name": "agentpool1",         "count": 3,         "vmSize": "Standard_D2_v3",         "osType": "Linux"       }     ],     "linuxProfile": {       "adminUsername": "azureuser",       "ssh": {         "publicKeys": [           {             "keyData": ""           }         ]       }     },     "servicePrincipalProfile": {       "clientId": "",       "secret": ""     }   } }
```

```
./bin/aks-engine generate /go/src/github.com/Azure/aks-engine/_output/kubernetes-northeurope-27576.json --output-directory /go/src/github.com/Azure/aks-engine/_output/kubernetes-northeurope-27576
2019/06/11 11:52:01 
Kubeconfig:/go/src/github.com/Azure/aks-engine/_output/kubernetes-northeurope-27576/kubeconfig/kubeconfig.northeurope.json

$ az group deployment create --name kubernetes-northeurope-27576 --resource-group kubernetes-northeurope-27576 --template-file /go/src/github.com/Azure/aks-engine/_output/kubernetes-northeurope-27576/azuredeploy.json --parameters /go/src/github.com/Azure/aks-engine/_output/kubernetes-northeurope-27576/azuredeploy.parameters.json
```

```
$ k get nodes -o json
2019/06/11 11:56:00 NAME                                 STATUS   ROLES    AGE   VERSION   INTERNAL-IP    EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION      CONTAINER-RUNTIME
k8s-agentpool1-28072136-vmss000000   Ready    agent    19s   v1.12.9   10.240.0.34    <none>        Ubuntu 16.04.6 LTS   4.15.0-1046-azure   docker://3.0.5
k8s-agentpool1-28072136-vmss000001   Ready    agent    20s   v1.12.9   10.240.0.65    <none>        Ubuntu 16.04.6 LTS   4.15.0-1046-azure   docker://3.0.5
k8s-agentpool1-28072136-vmss000002   Ready    agent    16s   v1.12.9   10.240.0.96    <none>        Ubuntu 16.04.6 LTS   4.15.0-1046-azure   docker://3.0.5
k8s-master-28072136-0                Ready    master   21s   v1.12.9   10.255.255.5   <none>        Ubuntu 16.04.6 LTS   4.15.0-1046-azure   docker://3.0.5
```

```
./bin/aks-engine scale \
        --subscription-id $SUBSCRIPTION_ID \
        --deployment-dir _output/$RESOURCE_GROUP \
        --location $REGION \
        --resource-group $RESOURCE_GROUP \
        --master-FQDN "$RESOURCE_GROUP.$REGION.cloudapp.azure.com" \
        --node-pool agentpool1 \
        --new-node-count 1 \
        --auth-method client_secret \
        --client-id ${CLIENT_ID} \
        --client-secret ${CLIENT_SECRET}
INFO[0000] validating...                                
INFO[0005] Name suffix: 28072136                        
INFO[0005] Starting ARM Deployment (kubernetes-northeurope-27576-1380518357). This will take some time... 
time="2019-06-11T12:15:52Z" level=info msg="Found no resources with type Microsoft.Network/routeTables in the template." source="scaling command line"
INFO[0080] Finished ARM Deployment (kubernetes-northeurope-27576-1380518357). Succeeded 
Agent pid 40
2019/06/11 12:17:16 Picking Random Region from list [northeurope]
2019/06/11 12:17:16 Picked Random Region:northeurope
```

```
 ./bin/aks-engine upgrade --force \
      --subscription-id $SUBSCRIPTION_ID \
      --deployment-dir _output/$RESOURCE_GROUP \
      --location $REGION \
      --resource-group $RESOURCE_GROUP \
      --upgrade-version $ver_target \
      --auth-method client_secret \
      --client-id ${CLIENT_ID} \
      --client-secret ${CLIENT_SECRET}
INFO[0009] Upgrading cluster with name suffix: 19460701 
time="2019-06-11T16:38:40Z" level=info msg="VM k8s-agentpool1-19460701-vmss_0 in VMSS k8s-agentpool1-19460701-vmss has a current version of 1.12.9 and a desired version of 1.13.7. Upgrading this node."
time="2019-06-11T16:38:41Z" level=info msg="Master VM name: k8s-master-19460701-0, orchestrator: 1.12.9 (MasterVMs)"
time="2019-06-11T16:38:42Z" level=info msg="Upgrading to Kubernetes version 1.13.7"
time="2019-06-11T16:38:42Z" level=info msg="Master nodes StorageProfile: ManagedDisks"
time="2019-06-11T16:38:42Z" level=info msg="Prepping master nodes for upgrade..."
time="2019-06-11T16:38:42Z" level=info msg="Resource count before running NormalizeResourcesForK8sMasterUpgrade: 10"
time="2019-06-11T16:38:42Z" level=info msg="Evaluating if agent pool: master, resource: [concat(variables('masterVMNamePrefix'), copyIndex(variables('masterOffset')))] needs to be removed"
time="2019-06-11T16:38:42Z" level=info msg="Evaluating if extension: [concat(variables('masterVMNamePrefix'), copyIndex(variables('masterOffset')),'/cse', '-master-', copyIndex(variables('masterOffset')))] needs to be removed"
time="2019-06-11T16:38:42Z" level=info msg="Evaluating if extension: [concat(variables('masterVMNamePrefix'), copyIndex(variables('masterOffset')), '/computeAksLinuxBilling')] needs to be removed"
time="2019-06-11T16:38:42Z" level=info msg="Resource count after running NormalizeResourcesForK8sMasterUpgrade: 10"
time="2019-06-11T16:38:42Z" level=info msg="Total expected master count: 1"
time="2019-06-11T16:38:42Z" level=info msg="Master nodes that need to be upgraded: 1"
time="2019-06-11T16:38:42Z" level=info msg="Master nodes that have been upgraded: 0"
time="2019-06-11T16:38:42Z" level=info msg="Starting upgrade of master nodes..."
time="2019-06-11T16:38:42Z" level=info msg="masterNodesInCluster: 1"
time="2019-06-11T16:38:42Z" level=info msg="Upgrading Master VM: k8s-master-19460701-0"
time="2019-06-11T16:38:42Z" level=info msg="fetching VM: kubernetes-australiaeast-9974/k8s-master-19460701-0"
time="2019-06-11T16:38:43Z" level=info msg="found nic name for VM (kubernetes-australiaeast-9974/k8s-master-19460701-0): k8s-master-19460701-nic-0"
time="2019-06-11T16:38:43Z" level=info msg="deleting VM: kubernetes-australiaeast-9974/k8s-master-19460701-0"
time="2019-06-11T16:38:43Z" level=info msg="waiting for vm deletion: kubernetes-australiaeast-9974/k8s-master-19460701-0"
time="2019-06-11T16:40:16Z" level=info msg="deleting nic: kubernetes-australiaeast-9974/k8s-master-19460701-nic-0"
time="2019-06-11T16:40:16Z" level=info msg="waiting for nic deletion: kubernetes-australiaeast-9974/k8s-master-19460701-nic-0"
time="2019-06-11T16:40:18Z" level=info msg="deleting managed disk: kubernetes-australiaeast-9974/k8s-master-19460701-0_OsDisk_1_eb716528511946febca8a8b90d04e86b"
time="2019-06-11T16:40:21Z" level=info msg="Master offset: 0"
time="2019-06-11T16:40:21Z" level=info msg="Master pool set count to: 1 temporarily during upgrade..."
INFO[0112] Starting ARM Deployment (master-19-06-11T16.40.21-1035687596). This will take some time... 
INFO[0154] Finished ARM Deployment (master-19-06-11T16.40.21-1035687596). Error: Code="DeploymentFailed" Message="At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details." Details=[{"code":"Conflict","message":"{\r\n  \"status\": \"Failed\",\r\n  \"error\": {\r\n    \"code\": \"ResourceDeploymentFailure\",\r\n    \"message\": \"The resource operation completed with terminal provisioning state 'Failed'.\",\r\n    \"details\": [\r\n      {\r\n        \"code\": \"NetworkingInternalOperationError\",\r\n        \"message\": \"Unknown network allocation error.\"\r\n      }\r\n    ]\r\n  }\r\n}"}] 
time="2019-06-11T16:41:03Z" level=info msg="Error creating upgraded master VM: k8s-master-19460701-0"
Error: upgrading cluster: Code="DeploymentFailed" Message="At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details." Details=[{"code":"Conflict","message":"{\r\n  \"status\": \"Failed\",\r\n  \"error\": {\r\n    \"code\": \"ResourceDeploymentFailure\",\r\n    \"message\": \"The resource operation completed with terminal provisioning state 'Failed'.\",\r\n    \"details\": [\r\n      {\r\n        \"code\": \"NetworkingInternalOperationError\",\r\n        \"message\": \"Unknown network allocation error.\"\r\n      }\r\n    ]\r\n  }\r\n}"}]
```

**Comment by victornoel** [comment]
I had the same problem here, but doing the upgrade using AKS and az cli (not aks-engine) from 1.12.8 to 1.13.7. The error and context is the same, so I suppose the problem has the same source.

- I had a cluster (1.12.8) that was created with one node pool of 1 node (1.12.8). I added to it another node pool of 2 nodes (1.12.8).
- I upgraded the control plane (cf https://docs.microsoft.com/en-us/azure/aks/upgrade-cluster) to 1.13.7 successfully.
- When I upgraded the first node pool (https://docs.microsoft.com/en-us/azure/aks/use-multiple-node-pools#upgrade-a-node-pool) to 1.13.7, I got the following error: `Deployment failed. Correlation ID: 053aebe6-5326-4537-8ebb-d972e125366e. Code='NetworkingInternalOperationError' Message='Unknown network allocation error.'`

By the way, everytime I have problem with the first node pool of a cluster, all the traffic to the cluster (from its public ip that LB service got assigned) is not arriving, is that expected?
